test/style/filter.test.cpp

```diff
@@ -115,6 +127,18 @@ TEST(Filter, PropertyExpression) {
     ASSERT_FALSE(filter("[\"==\", [\"get\", \"two\"], 4]", {{"two", int64_t(2)}}));
 }
 
+TEST(Filter, LegacyProperty) {
+    ASSERT_TRUE(filter("[\"<=\", \"two\", 2]", {{"two", int64_t(2)}}));
+    ASSERT_FALSE(filter("[\"==\", \"two\", 4]", {{"two", int64_t(2)}}));
+
+    ASSERT_FALSE(filter("[\"<=\", \"two\", \"2\"]", {{"two", int64_t(2)}}));
+    ASSERT_FALSE(filter("[\"==\", \"bool\", false]", {{"two", true}}));
+
+    ASSERT_TRUE(filter("[\"<=\", \"two\", \"2\"]", {{"two", std::string("2")}}));
+    ASSERT_FALSE(filter("[\"<\", \"two\", \"1\"]", {{"two", std::string("2")}}));
+    ASSERT_FALSE(filter("[\"==\", \"two\", 4]", {{"two", std::string("2")}}));
+}
```

Comment:
GL JS has tests for the expression version of `any` and `all` (to make sure that we're correctly parsing them as expressions as opposed to legacy filters): https://github.com/mapbox/mapbox-gl-js/blob/95bb4c0bcbf2da406da28795baf6a325c189cc04/test/unit/style-spec/feature_filter.test.js#L27-L37 -- would it make sense to add those?

Reply:
Addressed at: https://github.com/mapbox/mapbox-gl-native/pull/11610/commits/01de0a9f8ba19fd4431869d6336767369674716d#diff-4574a618412773ee8823c6637c5221a2